kafka/build_darwin.go

```diff
@@ -1,22 +1,13 @@
 // +build !dynamic
 
+
 // This file was auto-generated by librdkafka_vendor/bundle-import.sh, DO NOT EDIT.
 
 package kafka
 
-/*
-This file exists upstream, but I'm commenting it out (instead of deleting it) to avoid gratuitious
-merge conflicts when merging upstream into our branch.
-
-This is being removed to force a dynamic library build in all circumstances without having to have
-every individual library user build their app with -tags=dynamic
-
-
 // #cgo CFLAGS: -DUSE_VENDORED_LIBRDKAFKA -DLIBRDKAFKA_STATICLIB
-// #cgo LDFLAGS: ${SRCDIR}/librdkafka_vendor/librdkafka_darwin.a  -lz -lm -lsasl2 -ldl -lpthread
+// #cgo LDFLAGS: ${SRCDIR}/librdkafka_vendor/librdkafka_darwin.a  -lm -lsasl2 -ldl -lpthread -framework CoreFoundation -framework SystemConfiguration
 import "C"
 
 // LibrdkafkaLinkInfo explains how librdkafka was linked to the Go client
-const LibrdkafkaLinkInfo = "static darwin from librdkafka-static-bundle-v1.7.0.tgz"
-
- */
```

Comment:
Just having a play with how we might bring back the static linking here. I think we'll want to do this is a separate PR though.